### DISCUSSION TITLE
Thinking behind incremental selection maps

### DISCUSSION THREAD

**Discussion by nomasprime** [nvim-treesitter/nvim-treesitter#1218] [Q&A]
Working on my maps and wondered what the thinking was behind the following:

```
init_selection = "gnn",
node_incremental = "grn",
scope_incremental = "grc",
node_decremental = "grm",
```

Think originally written by @kyazdani42

**Comment by stsewd** [comment]
Personally I have them mapped to

```lua
    keymaps = {
      init_selection = "<CR>",
      scope_incremental = "<CR>",
      node_incremental = "<TAB>",
      node_decremental = "<S-TAB>",
    },

```

**Comment by kyazdani42** [comment]
these were actually written by @theHamsta so i cannot give you more insight sorry :) i remapped these in my config.

**Comment by theHamsta** [comment]
no, it was some compromise between @vigoux  @kyazdani42 in a PR discussion. It was because someone complained that the original mappings where overwriting something (orginal was `g<otherkey>`). So an additional key was added. I never understood the mappings I wasn't even member of the organisation when they where chosen.

I saw 
```lua
    keymaps = {
      init_selection = "<CR>",
      scope_incremental = "<CR>",
      node_decremental = "<BS>",
    },
```
in the gitter chat sometime and I am now using these mapping (mostly I don't use the feature at all)

**Comment by kyazdani42** [comment]
i think you might be right. We will need to change those defaults.

**Comment by stsewd** [comment]
My keymaps for this feature are:

```lua
  incremental_selection = {
    enable = true,
    keymaps = {
      init_selection = "<CR>",
      scope_incremental = "<CR>",
      node_incremental = "<TAB>",
      node_decremental = "<S-TAB>",
    },
  },

```

**Comment by vigoux** [comment]
The provided mappings are the defaults, it's up to the user to find those who suits him more.
Having @stsewd ones as default is too opinionated, let's keep them neutral.

No point in keeping that open as an issue, I am converting this as a discussion instead.
